nuxeo-features/nuxeo-platform-publisher/nuxeo-platform-publisher-core/src/main/java/org/nuxeo/ecm/platform/publisher/impl/service/PublisherServiceImpl.java

```diff
@@ -108,7 +110,9 @@ public void applicationStarted(ComponentContext context) {
                 if (completedAbruptly) {
                     TransactionHelper.setTransactionRollbackOnly();
                 }
-                TransactionHelper.commitOrRollbackTransaction();
+                if (started) {
+                    TransactionHelper.commitOrRollbackTransaction();
+                }
```

Comment:
What if `TransactionHelper.isTransactionActive() == false`? `started` will value false whereas we called `TransactionHelper.startTransaction()`. Don't we miss a final `TransactionHelper.commitOrRollbackTransaction()` in that case?